### PR TITLE
feat: add more image and install pre task and pipeline

### LINKED ISSUE #36: Add some image to installer
@Abirdcfly Great! To support chaincodebuild pipelines ,we need some other images:
- Image for Git-clone task:  gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.21.0
- Image for Minio-fetch task: docker.io/bitnami/minio-client:2023
- Images for Kaniko
   -  kaniko execute:  gcr.io/kaniko-project/executor:v1.5.1@sha256:c6166717f7fe0b7da44908c986137ecfeab21f31ec3992f6e128fff8a94be8a5
   - kaniko bashh: docker.io/library/bash:5.1.4@sha256:c523c636b722339f41b6a431b44588ab2f762c5de5ec3bd7964420ff982fb1d9

Btw.. maybe it's better to put `git-clone`\`kaniko` task yaml into installer\tekton\pipelines\task  which is just like `minio-fetch `

_Originally posted by @bjwswang in https://github.com/bestchains/installer/issues/13#issuecomment-1455358062_

### LINKED ISSUE #40: auto-generate accesskey/secretke after minio deployed

### LINKED ISSUE #41: expose ingress for minio service

### PR DESCRIPTION
**Pull request by Abirdcfly**
1. Replace image `bitnami/minio-client` with `minio-mc`.
   Compared to official images, Bitnami images has some additional issues: 
	- Not all steps are open source (we encountered problems with the arm64 version of the bitnami-ingress image in another project).
	- They add extra environment variables and scripts. For example, here `MINIO_SERVER_HOST` is not an official environment variable. see https://github.com/bitnami/containers/blob/43435015e679e30dcb72b7cf9bf111048b467f36/bitnami/minio/2023/debian-11/rootfs/opt/bitnami/scripts/libminioclient.sh#L65-L70

2. Cache all used images to the DockHub account `hyperledgerk8s` for convenient offline use.All images list in `function pre_load_image()` in https://github.com/bestchains/installer/blob/main/scripts/kind.sh (note: `docker pull hyperledgerk8s/tekton-pipeline-args-powershell:nanoserver-b6d5ff` only has `windows/amd64` OS/ARCH) Fix #36

3. Provide examples of `taskrun` and `pipelinerun` that can be validated on a `kind` cluster for all tekton `tasks` and `pipelines` used.

4. Support Kaniko build, add example.

5. Minio Update: 
   - Random username and password generated. （Stored in a secret with the same name and namespace as the minio helm release, default `secret/fabric-minio` in namespace: `baas-system`） Fix #40
   - Ingress enabled by default. Fix #41

6. Kind export port 80/443 for ingress by default.

7. support install minio or tekton without fabric-operator for `example-test` in fabric-operator.

8. Update document https://github.com/bestchains/installer/blob/main/tekton/README.md

9. use [shfmt](https://github.com/mvdan/sh) to reformats shell scripts https://github.com/bestchains/installer/blob/main/scripts/e2e.sh

### CODE REVIEW

File: scripts/e2e.sh [hunk 1]
```diff
@@ -102,4 +102,9 @@ if [[ ${WAIT_TEKTON_INSTALL} == "YES" ]]; then
 		sleep 5
 	done
 	echo "deploy tekton successfully"
+	if [[ ${INSTALL_PRE_TEKTON_TASK} == "YES" ]]; then
+	  echo "install pre defined tekton task and pipeline for chaincode..."
+	  find tekton -type f -name "*.yaml" ! -path "*/sample/*" | xargs -n 1 kubectl apply -f -
```
Comment (Abirdcfly) [inline scripts/e2e.sh:107]: apply to `default` or `baas-system` or other ns?

File: tekton/pipelines/chaincodebuild-kaniko.yaml [hunk 2]
```diff
@@ -47,26 +47,19 @@ spec:
...
       description: The name of the image containing the Docker tool
-      default: "gcr.io/kaniko-project/executor:v1.9.1"
+      default: "hyperledgerk8s/kaniko-executor:v1.9.1"
     - name: DOCKERFILE
       description: The path of the dockerfile to execute (optional)
       default: "./Dockerfile"
     - name: CONTEXT
       description: The path of the directory to use as context (optional)
       default: "."
-  results:
```
Comment (Abirdcfly) [inline tekton/pipelines/chaincodebuild-kaniko.yaml:63]: TODO

File: tekton/README.md [hunk 3]
```diff
@@ -120,9 +120,12 @@ reference here [Sample with source `git`](./pipelines/sample/sample_git.yaml)
...
-1. Sample for Task `minio-fetch`
+1. Sample for Tasks:
 
-- [Sample](./task/minio-fetch/sample/samplerun.yaml)
+- [`git-clone` Sample](./task/git-clone/sample/test-in-upstream.yaml)
+- [`minio-fetch` Sample](./task/minio-fetch/sample/samplerun.yaml)
+- [`docker-build` Sample](./task/docker-build/sample/test-in-upstream.yaml)
+- [`kaniko` Sample](./task/docker-build/sample/kaniko.yaml)
 
 2. Sample for Pipeline `chaincodebuild`
```
Comment (Abirdcfly) [inline tekton/README.md:130]: TODO

**Comment by Abirdcfly** [comment]
TODO minio 用户名密码随机
TODO minio ingress
TODO git minio docker 构建例子。

**Comment by Abirdcfly** [comment]
TODO issue dockerbuild 不支持docker认证，是否需要添加。

**Comment by bjwswang** [comment]
> 

you mean image build with task dockerbuild ?

**Comment by Abirdcfly** [comment]
@bjwswang @0xff-dev @Lanture1064 for review. PTAL. Thanks.

**Review by bjwswang — APPROVED** [review]
/lgtm

**Review by 0xff-dev — APPROVED** [review]

**Pull request merged**